src/Temporalio.Extensions.Hosting/TemporalWorkerClientUpdateSubscriber.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Temporalio.Worker;
+
+namespace Temporalio.Extensions.Hosting
+{
+    /// <summary>
+    /// Subscribes to the <see cref="TemporalWorkerClientUpdater"/>, and propagates the worker client changes to the Temporal worker.
+    /// </summary>
+    public class TemporalWorkerClientUpdateSubscriber : IDisposable

```

Comment:
Arguably does not need to be public and arguably not needed (I think a try/finally in the worker is enough, though would recommend the disposable approach if this were public API)